crates/bevy_winit/src/system.rs

```diff
@@ -164,12 +164,52 @@ pub(crate) fn changed_window(
                     winit_window.set_fullscreen(new_mode);
                 }
             }
+
             if window.resolution != cache.window.resolution {
+                // If `scale_factor_override` changed, the constraints need to be updated with the size.
+                if window.resolution.scale_factor_override()
+                    != cache.window.resolution.scale_factor_override()
+                {
+                    let (min_inner_size, max_inner_size) = get_inner_size_constraints(&window);
+
+                    winit_window.set_min_inner_size(Some(min_inner_size));
+                    winit_window.set_max_inner_size(max_inner_size);
+                }
+
+                // Winit doesn't check constraints when using `set_inner_size`.
+                // Check them manually.
+                let constraints = window.resize_constraints.check_constraints();
+                let logical_width = if window.resolution.width() > constraints.max_width {
+                    constraints.max_width
+                } else if window.resolution.width() < constraints.min_width {
+                    constraints.min_width
+                } else {
+                    window.resolution.width()
+                };
+                let logical_height = if window.resolution.height() > constraints.max_height {
+                    constraints.max_height
+                } else if window.resolution.height() < constraints.min_height {
+                    constraints.min_height
+                } else {
+                    window.resolution.height()
+                };
+                window.resolution.set(logical_width, logical_height);
+
+                // Use `set_inner_size` to set the new size.
                 let physical_size = PhysicalSize::new(
                     window.resolution.physical_width(),
                     window.resolution.physical_height(),
                 );
                 winit_window.set_inner_size(physical_size);
+
+                // Winit doesn't send a WindowEvent::Resized when using `set_inner_size`.
```

Comment:
Actually I'm not sure if:
- that's always the case (like this comment says),
- that happens only when the window effectively doesn't resize (which is the case when it already reached fullscreen and you try to make it bigger than fullscreen again)

Even if it's the second case it doesn't matter much we just "do something useless"